src/api/simplybook/simplybook-api.ts

```diff
@@ -0,0 +1,78 @@
+import { Logger } from '@nestjs/common';
+import axios from 'axios';
+
+import { simplybookCompanyName, simplybookCredentials } from 'src/utils/constants';
+
+type Booking = {
+  client: {
+    name: string;
+    email: string;
+  };
+  code: string;
+};
+
+type SimplybookBookingInfo = {
+  clientEmail: string;
+  bookingCode: string;
+  date: Date;
+};
+
+const DATE_FORMAT_LENGTH = 'YYYY-mm-dd'.length;
+const SIMPLYBOOK_API_BASE_URL = 'https://user-api-v2.simplybook.me/admin';
+const LOGGER = new Logger('SimplybookAPI');
+
+const getAuthToken: () => Promise<string> = async () => {
+  try {
+    const response = await axios({
+      method: 'post',
+      url: `${SIMPLYBOOK_API_BASE_URL}/auth`,
+      data: JSON.parse(simplybookCredentials),
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    });
+
+    return response.data.token;
+  } catch (error) {
+    LOGGER.error('Failed to authenticate against Simplybook API.', error);
+
+    throw error;
+  }
+};
+
+const getBookingsForDate: (date: Date) => Promise<Booking[]> = async (date: Date) => {
+  const token = await getAuthToken();
+
+  const simplybookFilterDateString = date.toISOString().substring(0, DATE_FORMAT_LENGTH);
+
+  try {
+    const bookingsResponse = await axios({
+      method: 'get',
+      url: `${SIMPLYBOOK_API_BASE_URL}/bookings?filter[date]=${simplybookFilterDateString}&filter[status]=confirmed`,
+      headers: {
+        'Content-Type': 'application/json',
+        'X-Company-Login': simplybookCompanyName,
+        'X-Token': `${token}`,
+      },
+    });
+
+    return bookingsResponse.data.data;
+  } catch (error) {
+    LOGGER.error(
+      `Failed to retrieve client booking information for ${date} from Simplybook API.`,
+      error,
+    );
+    throw error;
```

Comment:
I think if you throw the error in the catch it gets propagated down to getTherapyBookingInfoForDate. Which in turn has no error handling. 
I was just looking at this article for typescript approaches. 
https://blog.logrocket.com/improve-error-handling-typescript-exhaustive-type-checking/#popular-typescript-error-handling-approaches

I guess just work out what happens to getTherapyBookingInfoForDate if that error happens?

Reply:
I'm happy to propegate down in try catches if that seems the best way. I've done null checks before but that has it's flaws.